include/mbedtls/platform_util.h

```diff
@@ -41,6 +41,34 @@
 extern "C" {
 #endif
 
+#if defined( MBEDTLS_CHECK_PARAMS ) && defined(MBEDTLS_PLATFORM_C) && \
```

Comment:
I don't think this should be guarded by `MBEDTLS_PLATFORM_C`; we want a declaration of `MBEDTLS_PARAM_FAILED` and `mbedtls_param_failed` regardless of whether the platform module is enabled.